buildifier/buildifier.go

```diff
@@ -175,6 +175,9 @@ func main() {
 	}
 	diff = differ
 
+	tf := &utils.TempFile{}
+	defer tf.Clean()
```

Comment:
defers are not run in case of os.Exit, only when a func returns

Reply:
Fixed. Although `tf.Clean()` may be not called because of `os.Exit()`, it doesn't make sense to call it anyway if `processFile`(`s`) hasn't been called.

Reply:
maybe the answer is to make main() a lot shorter and move most of this code into a 'run' method that has this defer, and also a 'return exitCode', where only main calls os.Exit(exitCode)

Reply:
Done. `main` is not a lot shorter though, but it only contains parsing and validating of the arguments and a call to `run()` now.